### PR TITLE
SHOP-7: Changed registration, added validation for registration

### PR DESCRIPTION
**Pull request by Wraan**
small changes, cleaning

### CODE REVIEW

File: src/main/java/com/shop/service/UserServiceImpl.java [hunk 1]
```diff
@@ -26,22 +28,29 @@ public UserServiceImpl(UserRepository userRepository, RoleRepository roleReposit
         this.passwordEncoder = passwordEncoder;
     }
 
-    @Override
-    public User createUserFromForm(RegistrationDto registrationDto) {
-        return new User(Calendar.getInstance(), registrationDto.getUsername(), registrationDto.getPassword(), registrationDto.getEmail());
+
```
Comment (Swistak06) [inline src/main/java/com/shop/service/UserServiceImpl.java]: Unnecessary line.

File: src/main/java/com/shop/service/UserServiceImpl.java [hunk 2]
```diff
@@ -51,4 +60,5 @@ public void addUserRole(User user) {
     public User findByUsername(String username) {
         return userRepository.findByUsername(username);
     }
+
```
Comment (Swistak06) [inline src/main/java/com/shop/service/UserServiceImpl.java]: Unnecessary line

File: src/main/resources/static/js/signup-validate.js [hunk 3]
```diff
@@ -1,9 +1,60 @@
...
+    if(!validateItem($("#username"), new RegExp("([a-zA-Z0-9]{6,32}$)"),
+            "You can use a-z, A-Z, 0-9 and in range 6-32 characters."))
+        return false;
+    if(!validateItem($("#email"), new RegExp(".+\\@.+\\..+")
+            ,"Wrong email address."))
+        return false;
+    if(!checkBothPasswords())
+        return false;
+    if(!validateItem($("#pass"), new RegExp("([a-zA-Z0-9]{6,32}$)"),
+            "You can use a-z, A-Z, 0-9 and in range 6-32 characters."))
```
Comment (polskikiel) [inline src/main/resources/static/js/signup-validate.js]: switch this message to:  password should contain at least 8 characters (including upper and lower case letters and digits).
And do the same to the others bellow.

File: src/main/java/com/shop/controller/RegisterController.java [hunk 4]
```diff
@@ -20,13 +21,18 @@
...
+        model.addAttribute("register", new RegistrationDto());
         return "signUp";
     }
 
     @PostMapping("/signUp")
-    public String signUpNewUser(@ModelAttribute("user") @Valid RegistrationDto registrationDto) {
-        userService.register(userService.createUserFromForm(registrationDto));
+    public String signUpNewUser(@ModelAttribute("register") @Valid RegistrationDto registrationDto,
+                                BindingResult result) {
+        //or redirect to error page, we will discuss it maybe later
```
Comment (polskikiel) [inline src/main/java/com/shop/controller/RegisterController.java]: redirect:/singup?err

?

File: src/main/java/com/shop/dto/AddressDto.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    private String firstname;
+    @Length(max = 255)
+    private String lastname;
+    @Length(max = 255)
+    private String city;
+    @Length(max = 255)
+    private String street;
+    @Pattern(regexp = "([0-9]{2}-[0-9]{3}$)")
+    private String zipCode;
+    @Pattern(regexp = "([0-9]{9}$)")
```
Comment (polskikiel) [inline src/main/java/com/shop/dto/AddressDto.java]: add **@NotBlank** everywhere

File: src/main/java/com/shop/service/UserServiceImpl.java [hunk 6]
```diff
@@ -26,22 +28,28 @@ public UserServiceImpl(UserRepository userRepository, RoleRepository roleReposit
...
-        return new User(Calendar.getInstance(), registrationDto.getUsername(), registrationDto.getPassword(), registrationDto.getEmail());
+    public User createUserFromRegistrationDto(RegistrationDto registrationDto) {
+        User user = new User(Calendar.getInstance(), registrationDto.getUsername().trim(),
+                registrationDto.getPassword().trim(), registrationDto.getEmail().trim());
+        Address address = createAddressFromAddressDto(user, registrationDto.getAddressDto());
+        user.getAddresses().add(address);
+        return user;
+    }
+
+    private Address createAddressFromAddressDto(User user, AddressDto addressDto){
```
Comment (polskikiel) [inline src/main/java/com/shop/service/UserServiceImpl.java]: Move this method to address service, or we can do sth like dtoService

File: src/main/java/com/shop/service/UserServiceImpl.java [hunk 7]
```diff
@@ -26,22 +28,28 @@ public UserServiceImpl(UserRepository userRepository, RoleRepository roleReposit
...
+        return new Address(user, addressDto.getFirstname().trim(), addressDto.getLastname().trim(), addressDto.getCity().trim(),
+                addressDto.getStreet().trim(), addressDto.getZipCode().trim(), addressDto.getPhone().trim());
     }
 
     @Override
     public User register(User user) {
         user.setEnabled(true);
-        user.setRegistrationDate(Calendar.getInstance());
```
Comment (polskikiel) [inline src/main/java/com/shop/service/UserServiceImpl.java]: Why did u remove this line?

**Review by polskikiel — CHANGES_REQUESTED** [review]

File: src/main/java/com/shop/service/UserServiceImpl.java (hunk 7 above)
Comment (Wraan) [reply src/main/java/com/shop/service/UserServiceImpl.java]: We dont need this, because it's in User constructor

**Review by polskikiel — APPROVED** [review]

**Pull request merged**